owncloudApp/src/androidTest/java/com/owncloud/android/flow/DeletePublicShareTest.kt

```diff
@@ -0,0 +1,309 @@
+import com.owncloud.android.data.sharing.shares.db.OCShareEntity
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.lib.common.accounts.AccountUtils
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.status.CapabilityBooleanType
+import com.owncloud.android.lib.resources.status.OwnCloudVersion
+import com.owncloud.android.presentation.viewmodels.capabilities.OCCapabilityViewModel
+import com.owncloud.android.presentation.viewmodels.sharing.OCShareViewModel
+import com.owncloud.android.presentation.ui.sharing.ShareActivity
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.utils.AccountsManager
+import com.owncloud.android.utils.AppTestUtil
+import io.mockk.every
+import io.mockk.mockk
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.not
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.koin.android.ext.koin.androidContext
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
+
+class DeletePublicShareTest {
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(
+        ShareActivity::class.java,
+        true,
+        false
+    )
+
+    private lateinit var file: OCFile
+
+    private val publicShares = arrayListOf(
+        AppTestUtil.createPublicShare( // With no expiration date
+            path = "/Photos/image.jpg",
+            expirationDate = 0L,
+            isFolder = false,
+            name = "image.jpg link",
+            shareLink = "http://server:port/s/1"
+        ),
+        AppTestUtil.createPublicShare(
+            path = "/Photos/image.jpg",
+            isFolder = false,
+            name = "image.jpg updated link",
+            shareLink = "http://server:port/s/2"
+        )
+    )
+
+    private val capabilitiesLiveData = MutableLiveData<DataResult<OCCapabilityEntity>>()
+    private val sharesLiveData = MutableLiveData<DataResult<List<OCShareEntity>>>()
+
+    private val ocCapabilityViewModel = mockk<OCCapabilityViewModel>(relaxed = true)
+    private val ocShareViewModel = mockk<OCShareViewModel>(relaxed = true)
+
+    companion object {
+        private val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
+        private val account = Account("admin", "owncloud")
+
+        @BeforeClass
+        @JvmStatic
+        fun init() {
+            addAccount()
+        }
+
+        @AfterClass
+        @JvmStatic
+        fun cleanUp() {
+            AccountsManager.deleteAllAccounts(targetContext)
+        }
+
+        private fun addAccount() {
+            // obtaining an AccountManager instance
+            val accountManager = AccountManager.get(targetContext)
+
+            accountManager.addAccountExplicitly(account, "a", null)
+
```

Comment:
Why don't you create a class to be extended for instrumented tests with these functions?? I have seen this code repeated several times....